bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -141,9 +152,25 @@ public Builder setStatsLogger(StatsLogger statsLogger) {
             return this;
         }
 
+
+        public Builder dnsResolver(DNSToSwitchMapping dnsResolver) {
+            this.dnsResolver = dnsResolver;
+            return this;
+        }
+
+        public Builder requestTimer(HashedWheelTimer requestTimer) {
```

Comment:
Matteo is merging Yahoo's netty 4 change back. We can address the concern about public API and netty dependency there.